docs/design/y-flow/y-flow-nb-api.md

```diff
@@ -0,0 +1,350 @@
+        "inner_vlan_id": 0
+      },
+      "status": "IN_PROGRESS",
+      "description": "description",
+      "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+      "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]"
+    }
+  ],
+  "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+  "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]",
+}
+```
+
+## Read operation
+
+REST endpoint: `GET /v2/y-flows/{y_flow_id}`
+
+response payload:
+
+```json
+{
+  "y_flow_id": "yf-AAAAAAAAAAAAAAA",
+  "status": "UP",
+  "shared_endpoint": {
+    "switch_id": "00:00:00:00:00:00:00:01",
+    "port_number": 0
+  },
+  "maximum_bandwidth": 0,
+  "path_computation_strategy": "strategy-name",
+  "encapsulation_type": "encapsulation-name",
+  "max_latency": 0,
+  "max_latency_tier2": 0,
+  "ignore_bandwidth": true,
+  "periodic_pings": false,
+  "pinned": false,
+  "priority": 0,
+  "strict_bandwidth": true,
+  "description": "description",
+  "y_point": "00:00:00:00:00:00:01:01",
+  "sub_flows": [
+    {
+      "flow_id": "fAAAAAAAAAAAAAAA",
+      "endpoint": {
+        "switch_id": "00:00:00:00:00:00:00:02",
+        "port_number": 0,
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "shared_endpoint": {
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "status": "IN_PROGRESS",
+      "description": "description",
+      "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+      "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]"
+    },
+    {
+      "flow_id": "fAAAAAAAAAAAAAAB",
+      "endpoint": {
+        "switch_id": "00:00:00:00:00:00:00:03",
+        "port_number": 0,
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "shared_endpoint": {
+        "vlan_id": 0,
+        "inner_vlan_id": 0
+      },
+      "status": "IN_PROGRESS",
+      "description": "description",
+      "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+      "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]"
+    }
+  ],
+  "time_create": "YYYY-MM-DDThh:mm:ss[.SSS]",
+  "time_update": "YYYY-MM-DDThh:mm:ss[.SSS]",
+}
+```
+
+`y_point` contain switch_id where flow path's become different. I.e. from "shared_endpoint" to "y_point" all subordinate
+flows have same path, from "y_point" to each "endpoint" defined in "flows" array they have independent path's. Can
+be `null` if can't be determined (whole y_flow status in this case can't be UP).
+
+`sub_flows` contain info about "second"(not shared) endpoint of subordinate flows. It can contain from 0 to 2 records.
+
+`status` calculated from subordinate flow's statuses. It will `UP` if all subordinate flow statuses are `UP`. It will
+be `DEGRADED` if any of subordinate flow status is not `UP`. It will be `DOWN` if there is no one subordinate flow
+in `UP` status or there is no any subordinate flow at all. At least one subordinate flow must be `UP` for y_flow start
+passing traffic.
+
+## Update operation
+
+Changes will be propagated on all already existing subordinate flows.
+
+REST endpoints: 
+ * `PUT /v2/y-flows/{y_flow_id}`
```

Comment:
I'd propose not to implement put at all, since client could use patch, which is even more flexible. What do you think?

Reply:
They will have shared hub, service and FSM. It's easy to support both in such case.